File: app/Http/Controllers/Payment/ShowCheckoutController.php
<?php

namespace App\Http\Controllers\Payment;

use App\Http\Controllers\Controller;
use Illuminate\View\View;

class ShowCheckoutController extends Controller
{
    /**
     * Display the checkout page.
     */
    public function __invoke(string $checkoutId): View
    {
        $amount = session('checkout_amount', 0.00);
        $description = session('checkout_description', 'Payment');

        return view('payment.checkout', [
            'checkoutId' => $checkoutId,
            'amount' => $amount,
            'description' => $description,
        ]);
    }
}

File: app/controllers/payment/show_checkout_controller.py
from flask import render_template, session
from flask_login import login_required


class ShowCheckoutController:
    @login_required
    def __call__(self, checkout_id):
        amount = session.get("checkout_amount", 100.00)
        description = session.get("checkout_description", "Payment")

        return render_template(
            "payment/checkout.html",
            checkout_id=checkout_id,
            amount=amount,
            description=description,
        )
